src/CMakeLists.txt

```diff
@@ -298,24 +298,48 @@ endif()
 # Installation
 #
 
-include(GNUInstallDirs)
+if(NOT OPENVINO_TOKENIZERS_INSTALL_LIBDIR AND NOT OPENVINO_TOKENIZERS_INSTALL_BINDIR)
+  # - Windows: `<openvino_dir>\runtime\bin\intel64\Release\`
+  # - MacOS_x86: `<openvino_dir>/runtime/lib/intel64/Release`
+  # - MacOS_arm64: `<openvino_dir>/runtime/lib/arm64/Release/`
+  # - Linux_x86: `<openvino_dir>/runtime/lib/intel64/`
+  # - Linux_arm64: `<openvino_dir>/runtime/lib/aarch64/`
+  string(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" OPENVINO_TOKENIZERS_INSTALL_DIR)
+  if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*")
+    set(OPENVINO_TOKENIZERS_INSTALL_DIR intel64)
+  elseif(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "^(arm64.*|aarch64.*|AARCH64.*|ARM64.*)")
+    if(APPLE)
+      set(OPENVINO_TOKENIZERS_INSTALL_DIR "arm64")
+    else()
+      set(OPENVINO_TOKENIZERS_INSTALL_DIR "aarch64")
+    endif()
+  elseif(OPENVINO_TOKENIZERS_INSTALL_DIR STREQUAL "x86_64" OR OPENVINO_TOKENIZERS_INSTALL_DIR STREQUAL "amd64"  # Windows detects Intel's 64-bit CPU as AMD64
+      OR CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
+    set(OPENVINO_TOKENIZERS_INSTALL_DIR intel64)
+  endif()
+  if(MSVC OR APPLE)
+    set(OPENVINO_TOKENIZERS_INSTALL_DIR ${OPENVINO_TOKENIZERS_INSTALL_DIR}/${CMAKE_BUILD_TYPE})
+  endif()
+  set(OPENVINO_TOKENIZERS_INSTALL_LIBDIR runtime/lib/${OPENVINO_TOKENIZERS_INSTALL_DIR})
+  set(OPENVINO_TOKENIZERS_INSTALL_BINDIR runtime/bin/${OPENVINO_TOKENIZERS_INSTALL_DIR})
```

Comment:
can we put these vars to cache? so, users can see it in cmake GUI / CMakeCache.txt and knows that he / she can override it. Otherwise, they are hidden internally

It will also eliminate necessity to check `if(NOT OPENVINO_TOKENIZERS_INSTALL_LIBDIR AND NOT OPENVINO_TOKENIZERS_INSTALL_BINDIR)`